docs/tutorials/multimodal/text2text_matching.md

```diff
@@ -0,0 +1,79 @@
+
+## Prepare your Data
+In this tutorial, we will demonstrate how to use AutoMM for text-to-text matching with the Stanford Natural Language Inference ([SNLI](https://nlp.stanford.edu/projects/snli/)) corpus. SNLI is a corpus contains around 570k human-written sentence pairs labeled with *entailment*, *contradiction*, and *neutral*. It is a widely used benchmark for evaluating the representation and inference capbility of machine learning methods. The following table contains three examples taken from this corpus.
+
+| Premise                                                   | Hypothesis                                                           | Label         |
+|-----------------------------------------------------------|----------------------------------------------------------------------|---------------|
+| A black race car starts up in front of a crowd of people. | A man is driving down a lonely road.                                 | contradiction |
+|  An older and younger man smiling.                        | Two men are smiling and laughing at the cats playing on the   floor. | neutral       |
+| A soccer game with multiple males playing.                | Some men are playing a sport.                                        | entailment    |
+
+Here, we consider sentence pairs with label *entailment* as positive pairs (labeled as 1) and those with label *contradiction* as negative pairs (labeled as 0). Sentence pairs with neural relationship are discarded. The following code downloads and loads the corpus into dataframes.
+
+```{.python .input}
+from autogluon.core.utils.loaders import load_pd
+import pandas as pd
+
+snli_train = load_pd.load('https://automl-mm-bench.s3.amazonaws.com/snli/snli_train.csv', delimiter="|")
+snli_test = load_pd.load('https://automl-mm-bench.s3.amazonaws.com/snli/snli_test.csv', delimiter="|")
+snli_train.head()
+```
+
+## Train your Model
+
+Ideally, we want to obtain a model that can return high/low scores for positive/negative text pairs. Traditional text similarity methods only work on a lexical level without taking the semantic aspect into account, for example, using term frequency or tf-idf vectors. With AutoMM, we can easily train a model that captures the semantic relationship between sentences. Bascially, it uses [BERT](https://arxiv.org/abs/1810.04805) to project each sentence into a high-dimensional vector and treat the matching problem as a classification problem following the design in [sentence transformers](https://www.sbert.net/). 
+
+With AutoMM, you just need to specify the query, response, and label column names and fit the model on the training dataset without worrying the implementation details.
+
+```{.python .input}
+from autogluon.multimodal import MultiModalPredictor
+from autogluon.multimodal.constants import BINARY, MULTICLASS, QUERY, RESPONSE, UNIFORM_SOUP
+
+# Initialize the model
+matcher = MultiModalPredictor(
```

Comment:
`pipeline="text_similarity"` is missing. Without this, it would use predictor instead of matcher.

Reply:
Shouldn’t pipeline be redirected or removed?

Reply:
I think the refactor is not done yet. For now `pipeline` is still used in the master branch.